src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -54,7 +58,14 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     if (params) {
       payload = merge(payload, params);
     }
-    payload = { ...payload, step };
+    if (isActionStep) {
+      payload = {
+        ...payload,
+        actions: [{ name: step, params }],
+      };
```

Comment:
This will break all functionality on rain/monolith

Reply:
have you verified it locally, in the end of this function, we have a step to add stateToken to params

Reply:
Oh I see it, was hidden. Ty!